packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -210,15 +341,32 @@ export class CanvasManager implements CanvasManagerInterface {
     let lastSnapshotTime = 0;
     let rafId: number;
 
-    const getCanvas = (): HTMLCanvasElement[] => {
+    const getCanvas = (
+      canvasElement?: HTMLCanvasElement,
+    ): HTMLCanvasElement[] => {
       const matchedCanvas: HTMLCanvasElement[] = [];
-      win.document.querySelectorAll('canvas').forEach((canvas) => {
+      if (canvasElement) {
         if (
-          !isBlocked(canvas, blockClass, blockSelector, unblockSelector, true)
+          !isBlocked(
+            canvasElement,
+            blockClass,
+            blockSelector,
+            unblockSelector,
+            true,
+          )
```

Comment:
I don't think this is needed? If `canvasElement` is always supplied by the user (can you confirm this?), then we can assume that the user is not passing in elements that should be blocked.

Reply:
yes, canvasElement is always supplied by the user